orbit/pkg/update/notifications.go

```diff
@@ -83,6 +88,22 @@ func (h *renewEnrollmentProfileConfigFetcher) GetConfig() (*fleet.OrbitConfig, e
 					return cfg, nil
 				}
 
+				// we perform this check locally on the client too to avoid showing the
+				// dialog if the Fleet enrollment profile has not been assigned to the device in
+				// Apple Business Manager.
+				assignedFn := h.checkAssignedEnrollmentProfileFn
+				if assignedFn == nil {
+					assignedFn = profiles.CheckAssignedEnrollmentProfile
+				}
```

Comment:
just in case: all the errors here are because you need to implement the matching function in `orbit/pkg/profiles/profiles_notdarwin.go`